docs/design/password-key-derivation.md

```diff
@@ -26,51 +26,52 @@ The third requirement allows having multiple encrypted domains to exist within a
 
 ## Design
 
-Uplink always stores derived keys, never the input material.
+Uplink always stores and operates on outputs of the following algorithms, never the input material.
 
-The satellite will add a new endpoint, `GetSaltEntropy` that uses the api key to find the project, and optionally takes a bucket name. It is important that the result is stable, even if the bucket is renamed. For example the result of `GetSaltEntropy("bucket1")` should be the same as `GetSaltEntropy("bucket2")` even after `"bucket1"` is renamed to `"bucket2"`. Note that there are still significant challenges to bucket renaming, and we don't support it right now.
+Satellites will add a salt field to bucket metadata that lives with the path encryption information. Satellites will also store a salt that lives with the project metadata. The salts will be 32 random, stable bytes.
 
-When deriving a root key from an api key, password, optional bucket name and optional encrypted path, the following algorithm will be used
+A default password can be chosen for an api key. The following algorithm is used to derive the default password:
 
 ```
-saltEntropy = GetSaltEntropy(bucketName or None)
+saltEntropy     = getProjectSalt(apiKey)
+mixedSalt       = hmac(hash=sha256, secret=userPassword, data=saltEntropy)
+defaultPassword = argon2id(salt=mixedSalt, password=userPassword)
+```
+
+The following algorithm is used to deriving a root key from an api key, password, optional bucket name and optional encrypted path:
```

Comment:
Bucket name is no longer optional, if I'm understanding correctly.

Reply:
:+1: